lib/rubocop/cop/rails/where_missing.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Check for the use `left_joins` and `where` method for finding missing relationship records.
+      #
+      # This cop is enabled in Rails 6.1 or higher.
+      #
+      # @example
+      #   # bad
+      #   Foo.left_joins(:foo).where(foos: { id: nil })
+      #
+      #   # good
+      #   Foo.where.missing(:foo)
+      #
+      class WhereMissing < Base
+        include RangeHelp
+        extend AutoCorrector
+        extend TargetRailsVersion
+
+        MSG = 'Use `where.missing(:%<association>s)` instead of ' \
+              '`%<left_joins_method>s(:%<association>s).where(%<association>ss: { id: nil })`.'
+        RESTRICT_ON_SEND = %i[left_joins left_outer_joins].freeze
+
+        minimum_target_rails_version 6.1
+
+        # @!method where_arguments(node)
+        def_node_search :where_arguments, <<~PATTERN
+          {
+            (send ... :where (hash  (pair $(sym _) (hash (pair (sym :id) (nil)))) ... ))
+            (send ... :where (hash ... (pair $(sym _) (hash (pair (sym :id) (nil)))) ... ))
```

Comment:
I believe you can achieve this with single pattern like this
```suggestion
            (send ... :where (hash <(pair $(sym _) (hash (pair (sym :id) (nil))))...> ))
```

Reply:
Indeed it is. Thank you very much.
Make sure it is a single pattern.